packages/babel-generator/src/generators/statements.ts

```diff
@@ -89,11 +100,11 @@ export function WhileStatement(this: Printer, node: t.WhileStatement) {
   this.printBlock(node);
 }
 
-const buildForXStatement = function (op) {
-  return function (this: Printer, node: any) {
+const buildForXStatement = function (op: "in" | "of") {
```

Comment:
It's a small refactor, but can we remove this parameter and just check `node.type === "ForOfStatement" ? "of" : "in"`?